### PR TITLE
Add module

### PR DESCRIPTION
**Pull request by kharijarrett**
Obviously I'm still picking over it and making small changes, but I'd like your eyes on this when you get a chance.

### CODE REVIEW

File: src/aws_sagemaker/aws_sagemaker.py [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+                                 *, 
+                                 extra: Optional[Dict[str, Any]] = None,
+                                 timeout: Optional[float] = None,
+                                 **kwargs) -> List[Classification]:
+        classifications = []
+        if isinstance(image, RawImage):
+            response = self.client.invoke_endpoint(EndpointName=self.endpoint_name, 
+                                                   ContentType= 'application/x-image',
+                                                   Accept='application/json;verbose',
+                                                   Body=image.data) # send image.data[24:]??
```
Comment (bhaney) [inline src/aws_sagemaker/aws_sagemaker.py:119]: For RawImage, you need to check what kind of bytes these are first before sending them over. RawImage has a "data" and "mime_type" field. So first check if the mime_type is something you know how to handle before sending image.data in the body. RawImage does not always mean "viam.vnd.rgba"

File: src/aws_sagemaker/aws_sagemaker.py [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+            image.save(stream, "JPEG")
+            response = self.client.invoke_endpoint(EndpointName=self.endpoint_name, 
+                                                   ContentType= 'application/x-image',
+                                                   Accept='application/json;verbose',
+                                                   Body=stream.getvalue())
+            
+        # Either way... 
+        out = json.loads(response['Body'].read())
+        labels = out['labels']
+        probs = out['probabilities']
```
Comment (bhaney) [inline src/aws_sagemaker/aws_sagemaker.py:132]: Is it always going to be labels and probabilities?

File: src/aws_sagemaker/aws_sagemaker.py [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+    async def get_detections(self,
+                            image: Union[Image.Image, RawImage],
+                            *,
+                            extra: Optional[Dict[str, Any]] = None,
+                            timeout: Optional[float] = None,
+                            **kwargs) -> List[Detection]:
+        
+        detections = []
+        if isinstance(image, RawImage):
+            width, height = int.from_bytes(image.data[8:16], "big"), int.from_bytes(image.data[16:24], "big")
```
Comment (bhaney) [inline src/aws_sagemaker/aws_sagemaker.py:165]: before you do this, you have to check the mime_type field of the RawImage to know if you are dealing with a vnd.viam.rgba or another kind of image type.

File: src/aws_sagemaker/aws_sagemaker.py [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+                                                   ContentType= 'application/x-image',
+                                                   Accept='application/json;verbose',
+                                                   Body=stream.getvalue())
+            
+        # Either way... 
+        out = json.loads(response['Body'].read())
+        boxes =  out['normalized_boxes']
+        classes= out['classes']
+        scores = out['scores']
+        labels = out['labels']
```
Comment (bhaney) [inline src/aws_sagemaker/aws_sagemaker.py:185]: Will these fields be consistent across all models? potentially may need to make more config parameters to make the appropriate mapping

**Review by bhaney — COMMENTED** [review]
Here's the first pass of comments! You can do ahead and merge this PR into the repo since i know there will be other iterations after this

File: src/aws_sagemaker/aws_sagemaker.py (hunk 4 above)
Comment (kharijarrett) [reply src/aws_sagemaker/aws_sagemaker.py:185]: Came back just to say 100% of the pre-trained models Sagemaker offers have the exact same fields. Consistent for classifiers and consistent for detectors.

**Pull request merged**